File: CertificationsCtrl/src/main/java/br/sp/telesul/service/ExportService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package br.sp.telesul.service;

import br.sp.telesul.model.Funcionario;
import java.util.List;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.springframework.stereotype.Service;

/**
 *
 * @author ebranco
 */
@Service
public interface ExportService {

    public void buildExcelDocument(String type, String[] columns, String[] columnsFormacao, String[] columnsIdiomas, String[] columnsCertificacoes, String habilitado, HttpServletRequest request, HttpServletResponse response);
    public List<Funcionario> readExcelDocument();
    public void singleReport(Long id,HttpServletRequest request, HttpServletResponse response);

}

File: CertificationsCtrl/src/main/webapp/resources/angularjs/services/exportService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


var app = angular.module("rhApp");

app.constant("exportType", "export/exportFile"),
app.constant("exportSingleReport", "export/exportSingleReport"),
app.constant("readExcel", "export/readExcel").factory("exportService", function ($http, exportType,readExcel,exportSingleReport) {

    return{
        exportType: function (type, habilitado) {
            var ajaxget = $http({
                url: exportType + "/" + type + "/"+ habilitado,
                method: 'GET',
                async: true,
                cache: false,
                headers: {'Accept': 'application/json', 'Pragma': 'no-cache'}
            });
           
        },
        readExcel: function () {
            var ajaxget = $http({
                url: readExcel,
                method: 'GET',
                async: true,
                cache: false,
                headers: {'Accept': 'application/json', 'Pragma': 'no-cache'}
            });
           
        },
        exportSingleReport: function (id) {
            var ajaxget = $http({
                url: exportSingleReport+"/"+id,
                method: 'GET',
                async: true,
                cache: false,
                headers: {'Accept': 'application/json', 'Pragma': 'no-cache'}
            });
           
        }
    };

});

